src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -153,7 +155,13 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
                 if (translogId.v1() != null && skipInitialTranslogRecovery == false) {
                     recoverFromTranslog(translogId.v1(), transformer);
                 } else {
-                    flush(true, true);
+                    if (lastCommittedSegmentInfos.getUserData().get(SYNC_COMMIT_ID) == null) {
```

Comment:
can we add a nocommit here plus a comment as to why we do this? I think we need to think about this some more but don't want to stop development now..